web-console/src/views/home-view/datasources-card/datasources-card.tsx

```diff
@@ -50,14 +50,17 @@ export class DatasourcesCard extends React.PureComponent<
     this.datasourceQueryManager = new QueryManager({
       processQuery: async capabilities => {
         let datasources: string[];
-        if (capabilities !== 'no-sql') {
+        if (capabilities.hasSql()) {
           datasources = await queryDruidSql({
             query: `SELECT datasource FROM sys.segments GROUP BY 1`,
           });
-        } else {
+        } else if (capabilities.hasCoordinatorAccess()) {
           const datasourcesResp = await axios.get('/druid/coordinator/v1/datasources');
           datasources = datasourcesResp.data;
+        } else {
+          throw new Error(`must have SQL or coordinator access`);
```

Comment:
I see a handful of these similar themed errors, would it make sense to make some custom errors like `NoCoordinatorError`, `NoOverlordError`, etc so you don't have to repeat the string message?